src/pages/instances/forms/UploadInstanceFileModal.tsx

```diff
@@ -0,0 +1,89 @@
+import { FC, useState } from "react";
+import { Modal, RadioInput } from "@canonical/react-components";
+import { UploadState } from "types/storage";
+import ProgressBar from "components/ProgressBar";
+import { humanFileSize } from "util/helpers";
+import UploadInstanceBackupFileForm from "./UploadInstanceBackupFileForm";
+import UploadExternalFormatFileForm from "./UploadExternalFormatFileForm";
+import useEventListener from "@use-it/event-listener";
+import { useSupportedFeatures } from "context/useSupportedFeatures";
+
+const hasSufficientHeight = () => window.innerHeight >= 790;
+
+type InstanceFileType = "instance-backup" | "external-format";
+
+interface Props {
+  close: () => void;
+}
+
+const UploadInstanceFileModal: FC<Props> = ({ close }) => {
+  const [fileType, setFileType] = useState<InstanceFileType>("instance-backup");
+  const [uploadState, setUploadState] = useState<UploadState | null>(null);
+  const [fixedFormHeight, setFixedFormHeight] = useState(hasSufficientHeight());
+  const { hasInstanceImportConversion } = useSupportedFeatures();
+
+  // If the browser window has sufficient height, then display the form with fixed height
+  // This will prevent the form from resizing itself due to form content changing
+  // for smaller viewport height, the form will be scrollable and its height will be dynamic
+  useEventListener("resize", () => {
+    setFixedFormHeight(hasSufficientHeight());
+  });
+
+  return (
+    <Modal
+      close={close}
+      className="upload-instance-modal"
+      title="Upload instance"
+    >
+      {!uploadState && hasInstanceImportConversion && (
+        <>
+          <label htmlFor="file-type">Choose file type</label>
+          <div id="file-type" className="file-type">
+            <RadioInput
+              label="LXD backup archive (.tar.gz)"
+              inline
+              checked={fileType === "instance-backup"}
+              onChange={() => setFileType("instance-backup")}
+            />
+            <RadioInput
+              label="External format (.qcow2, .vmdk, etc...)"
+              inline
+              checked={fileType === "external-format"}
+              onChange={() => setFileType("external-format")}
+            />
```

Comment:
The styling looks a little busy, maybe placing them below each other will give this more order. Secondly, I think a bit more bottom spacing would be good for the two radios. See the `snapshot-schedule` CSS class for an example where we set some custom styling to make the radio placement a bit better.

Reply:
We can also use the `u-sv1` or up to `u-sv3` helper classes, that might be better than introducing more custom css.

Reply:
thanks I keep forgetting that we have these utility classes!